Source/UI/Search/SearchResultsView.swift

```diff
@@ -0,0 +1,239 @@
+        } catch {
+            Log.optional(error)
+            CrashReporting.shared.reportIfNeeded(error: error)
+            return []
+        }
+    }
+
+    /// Loads the message with the given id from the database and displays it if it's still valid.
+    func loadMessage(with msgID: MessageIdentifier) async -> Either<Message, MessageIdentifier> {
+        var result: Either<Message, MessageIdentifier>
+        do {
+            result = .left(try Bots.current.post(from: msgID))
+        } catch {
+            result = .right(msgID)
+            Log.optional(error)
+        }
+        return result
+    }
+}
+
+/// A model for all the different types of results that can be displayed.
+fileprivate struct SearchResults {
+    enum ResultData {
+        case universal(people: [About], posts: [Message])
+        case feedID(FeedIdentifier)
+        case messageID(Either<Message, MessageIdentifier>)
+        case loading
+        case none
+        var isReadyToSearch: Bool {
+            switch self {
+            case .none:
+                return true
+            default:
+                return false
+            }
+        }
+        var shouldShowLoading: Bool {
+            switch self {
+            case .none, .loading:
+                return true
+            default:
+                return false
+            }
+        }
+    }
+
+    var data: ResultData
+    var query: String
+
+    /// The table view sections that should be displayed for these results.
+    var activeSections: [SearchResultsSection] {
+        switch data {
+        case .universal(let users, _):
+            if users.isEmpty {
+                return [.all]
+            } else {
+                return [.all, .people]
+            }
+        case .feedID:
+            return [.all, .people]
+        case .messageID:
+            return [.all]
+        case .loading, .none:
+            return [.all]
+        }
+    }
+
+    var posts: [Message] {
+        switch data {
+        case .universal(_, let posts):
+            return posts
+        case .messageID(let result):
+            switch result {
+            case .left(let message):
+                return [message]
+            case .right:
+                return []
+            }
+        default:
+            return []
+        }
+    }
+
+    var users: [Either<FeedIdentifier, About>] {
+        switch data {
+        case .feedID(let identity):
+            return [.left(identity)]
+        case .universal(let abouts, _):
+            return abouts.map { Either<FeedIdentifier, About>.right($0) }
+        default:
+            return []
+        }
+    }
+
+    var isEmpty: Bool {
+        posts.isEmpty && users.isEmpty
+    }
+}
+
+struct SearchResultsView_Previews: PreviewProvider {
+    static var previews: some View {
+        SearchResultsView(searchText: "Hello")
+    }
+}
+
+extension Either: Identifiable, Equatable, Hashable where Left == FeedIdentifier, Right == About {
```

Comment:
I think this and `SearchResults` would be better in their own files.

Reply:
I moved it to the Models folder